src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -96,44 +117,40 @@ using namespace Containers::Literals;
 
 struct AssimpImporter::File {
     Containers::Optional<Containers::String> filePath;
+    UnsignedInt assimpVersion = 0;
     bool importerIsGltf = false;
     const aiScene* scene = nullptr;
     /* Index -> pointer, pointer -> index conversion for nodes as they're
        represented in a tree. Needed by objectCount() (which is const) so can't
        be delayed to scenne() parsing. */
-    std::vector<aiNode*> nodes;
+    Containers::Array<aiNode*> nodes;
     std::unordered_map<const aiNode*, UnsignedInt> nodeIndices;
     /* (materialPointer, propertyIndexInsideMaterial, imageIndex) tuple,
        imageIndex points to the (deduplicated) images array */
-    std::vector<std::tuple<const aiMaterial*, UnsignedInt, UnsignedInt>> textures;
+    Containers::Array<Containers::Triple<const aiMaterial*, UnsignedInt, UnsignedInt>> textures;
     /* (materialPointer, propertyIndexInsideMaterial) tuple defining the first
        (unique) location of an image */
-    std::vector<std::pair<const aiMaterial*, UnsignedInt>> images;
+    Containers::Array<Containers::Pair<const aiMaterial*, UnsignedInt>> images;
 
-    std::unordered_map<std::string, UnsignedInt> materialIndicesForName;
+    std::unordered_map<Containers::String, UnsignedInt> materialIndicesForName;
     std::unordered_map<const aiMaterial*, UnsignedInt> textureIndices;
 
-    Containers::Optional<std::unordered_map<std::string, Int>>
+    /* We can use StringView for the keys because they point to aiString mName
+       in the scene, which stay valid for as long as a file is opened */
+    Containers::Optional<std::unordered_map<Containers::StringView, Int>>
         animationsForName,
         camerasForName,
         lightsForName,
         meshesForName,
         skinsForName;
 
-    /* Joint ids and weights are the only custom attributes in this importer */
-    std::unordered_map<std::string, MeshAttribute> meshAttributesForName{
-        {"JOINTS", meshAttributeCustom(0)},
-        {"WEIGHTS", meshAttributeCustom(1)}
-    };
-    std::vector<std::string> meshAttributeNames{"JOINTS", "WEIGHTS"};
-
-    std::vector<std::size_t> meshesWithBones;
+    Containers::Array<std::size_t> meshesWithBones;
     /* For each mesh: the index of its skin (before any merging), or -1 */
-    std::vector<Int> meshSkins;
+    Containers::Array<Int> meshSkins;
     bool mergeSkins = false;
     /* For each mesh, map from mesh-relative bones to merged bone list */
-    std::vector<std::vector<UnsignedInt>> boneMap;
-    std::vector<const aiBone*> mergedBones;
+    Containers::Array<Containers::Array<UnsignedInt>> boneMap;
```

Comment:
Because the `boneMap` is filled sequentially (if I see correctly), I was wondering if we could avoid the nested allocations by having just one array and an offset array to index into that? If it's not too much extra testing effort.

Reply:
Good idea 👍  I'll check if the current tests cover this well enough, otherwise I'll extend them a bit

Reply:
I extended the tests a bit and I feel like I've hit a bug either with assimp or the bone merging logic in our code. Not too much time to investigate before I leave on vacation for ~2 weeks, but I'll pick this back up afterwards.

Reply:
In that case, can you just push/rebase the remaining changes except this one? This particular optimization can wait two weeks, but I now have 5.2.5 on my system as well (and I expect Homebrew will soon as well) and it'd be nice to have this merged to have tests passing again. 

Thank you!

Reply:
I pushed everything except this change. https://github.com/pezcode/magnum-plugins/tree/assimp-skin-merge-better-test has the change + extended tests that fail, if you care to look at it.

Reply:
Awesome, thanks :) Looked at the commit, but didn't see anything suspicious. Strange library, strange bugs.

Anyway, enjoy your vacation! :palm_tree: